### ISSUE TITLE
smooth on array with only Nan and edge_truncate results in wrong numbers

### ISSUE THREAD

**Issue by brandy125** [gnudatalanguage/gdl#1940]
Can we solve this?

```
GDL> a=fltarr(3,3)*'nan'
GDL> print,smooth(a,3,/nan,/edge_truncate)
  5.80707e-12          NaN -2.79620e+06
  2.90353e-12          NaN  2.40428e+22
  5.38529e-28          NaN  4.80856e+22
```

**Comment by GillesDuvert** [comment]
@brandy125 this comes from an out-of-bound condition... very bad bug.

**Comment by GillesDuvert** [comment]
closed in a few moments by #1944
